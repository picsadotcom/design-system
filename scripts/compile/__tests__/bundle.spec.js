// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/* eslint-env jest */

const Task = require("data.task");
const Bundle = require("../bundle");

describe("scripts/compile/bundle", () => {
  describe("configs", () => {
    it("umd", () => {
      expect(Bundle.configs.umd).toMatchSnapshot();
    });
    it("chunked", () => {
      expect.assertions(1);
      Bundle.configs.chunked.fork(
        e => {
          throw e;
        },
        config => {
          config = config.delete("plugins");
          expect(config).toMatchSnapshot();
        }
      );
    });
  });
  describe("compile", () => {
    it("rejects the task if the config contains no output.path", () => {
      expect.assertions(1);
      Bundle.compile(Bundle.configs.umd).fork(
        e => {
          expect(e.message).toContain("output.path");
        },
        r => {}
      );
    });
  });
  describe("createLibrary", () => {
    it("calls the compile function with the correct config", () => {
      expect.assertions(2);
      const compile = jest.fn(config => Task.of(config));
      Bundle.createLibrary("/testPath", compile).fork(
        e => {
          throw e;
        },
        configs => {
          configs = configs.map(c => c.delete("plugins"));
          expect(compile).toHaveBeenCalledTimes(2);
          expect(configs).toMatchSnapshot();
        }
      );
    });
  });
});
