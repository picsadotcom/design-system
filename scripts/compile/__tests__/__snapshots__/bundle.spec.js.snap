// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scripts/compile/bundle configs chunked 1`] = `
Object {
  "entry": Object {
    "components/buttons/base": "./ui/components/buttons/base/example.jsx",
    "components/buttons/stateful": "./ui/components/buttons/stateful/example.jsx",
    "components/buttons/with-icon": "./ui/components/buttons/with-icon/example.jsx",
    "components/data-tables/advanced": "./ui/components/data-tables/advanced/example.jsx",
    "components/data-tables/base": "./ui/components/data-tables/base/example.jsx",
    "components/data-tables/inline-edit": "./ui/components/data-tables/inline-edit/example.jsx",
    "utilities/alignment": "./ui/utilities/alignment/example.jsx",
    "utilities/floats": "./ui/utilities/floats/example.jsx",
  },
  "externals": Object {
    "react": "React",
    "react-dom": "ReactDOM",
  },
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "options": Object {
          "babelrc": false,
          "plugins": Array [
            "transform-object-rest-spread",
          ],
          "presets": Array [
            "es2015",
            "react",
          ],
        },
        "test": /\\\\\\.jsx\\?\\$/,
      },
    ],
  },
  "output": Object {
    "filename": "[name].js",
    "jsonpFunction": "webpackJsonpSLDS",
    "library": "SLDS",
    "publicPath": "/assets/scripts/bundle/",
  },
  "resolve": Object {
    "alias": Object {
      "lodash": "",
    },
    "extensions": Array [
      ".js",
      ".jsx",
    ],
  },
}
`;

exports[`scripts/compile/bundle configs umd 1`] = `
Object {
  "entry": "./scripts/compile/entry.js",
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "options": Object {
          "babelrc": false,
          "plugins": Array [
            "transform-object-rest-spread",
          ],
          "presets": Array [
            "es2015",
            "react",
          ],
        },
        "test": /\\\\\\.jsx\\?\\$/,
      },
    ],
  },
  "output": Object {
    "filename": "slds.umd.js",
    "library": "SLDS",
    "libraryTarget": "umd",
    "publicPath": "/assets/scripts/bundle/",
  },
  "plugins": Array [],
  "resolve": Object {
    "alias": Object {
      "lodash": "",
    },
    "extensions": Array [
      ".js",
      ".jsx",
    ],
  },
}
`;

exports[`scripts/compile/bundle createLibrary calls the compile function with the correct config 1`] = `
Array [
  Object {
    "entry": "./scripts/compile/entry.js",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "loader": "babel-loader",
          "options": Object {
            "babelrc": false,
            "plugins": Array [
              "transform-object-rest-spread",
            ],
            "presets": Array [
              "es2015",
              "react",
            ],
          },
          "test": /\\\\\\.jsx\\?\\$/,
        },
      ],
    },
    "output": Object {
      "filename": "slds.umd.js",
      "library": "SLDS",
      "libraryTarget": "umd",
      "path": "/testPath/__internal",
      "publicPath": "/assets/scripts/bundle/",
    },
    "resolve": Object {
      "alias": Object {
        "lodash": "",
      },
      "extensions": Array [
        ".js",
        ".jsx",
      ],
    },
  },
  Object {
    "entry": Object {
      "components/buttons/base": "./ui/components/buttons/base/example.jsx",
      "components/buttons/stateful": "./ui/components/buttons/stateful/example.jsx",
      "components/buttons/with-icon": "./ui/components/buttons/with-icon/example.jsx",
      "components/data-tables/advanced": "./ui/components/data-tables/advanced/example.jsx",
      "components/data-tables/base": "./ui/components/data-tables/base/example.jsx",
      "components/data-tables/inline-edit": "./ui/components/data-tables/inline-edit/example.jsx",
      "utilities/alignment": "./ui/utilities/alignment/example.jsx",
      "utilities/floats": "./ui/utilities/floats/example.jsx",
    },
    "externals": Object {
      "react": "React",
      "react-dom": "ReactDOM",
    },
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "loader": "babel-loader",
          "options": Object {
            "babelrc": false,
            "plugins": Array [
              "transform-object-rest-spread",
            ],
            "presets": Array [
              "es2015",
              "react",
            ],
          },
          "test": /\\\\\\.jsx\\?\\$/,
        },
      ],
    },
    "output": Object {
      "filename": "[name].js",
      "jsonpFunction": "webpackJsonpSLDS",
      "library": "SLDS",
      "path": "/testPath/__internal/chunked",
      "publicPath": "/assets/scripts/bundle/",
    },
    "resolve": Object {
      "alias": Object {
        "lodash": "",
      },
      "extensions": Array [
        ".js",
        ".jsx",
      ],
    },
  },
]
`;
